ruby_on_rails/send_emails.md

```diff
@@ -33,19 +21,68 @@ class ApplicationMailer < ActionMailer::Base
 end
 ```
 
-* add the following to `config/application.rb`
+- add the following to `config/application.rb`
 
 ```ruby
 config.action_mailer.default_url_options = { host: ENV['APP_HOST'], port: ENV['APP_PORT'] }
 ```
 
-* add the following to `config/environments/development.rb`:
+- add the following to `config/environments/development.rb`:
 
 ```ruby
 config.action_mailer.delivery_method = :letter_opener
 ```
 
-* add the following `config/environments/production.rb`:
+Now either setup SendGrid or SparkPost. If you need to send more than 100 mails per day use SparkPost otherwise use SendGrid. (It's more reliable and easier to set up.) (Sparkpost sometimes fails to send mails to outlook and bluewin email addresses.)
+
+## SendGrid
+
+- Add the SendGrid AddOn to Heroku
+
+- add the following to `config/application.example.yml`
+
+```yml
+APP_DOMAIN: '[project-name].localhost'
+APP_PORT: '3000'
+MAIL_SENDER: 'yourname+<application>@example.com'
+SENDGRID_USERNAME: ''
+SENDGRID_PASSWORD: ''
+```
+
+- Add this gem `sendgrid-ruby` to your gemfile and run `bundle`.
+
+- Add the following code to your `config/environments/production.rb`
+
+```sh
+ActionMailer::Base.smtp_settings = {
+  user_name: ENV['SENDGRID_USERNAME'],
+  password: ENV['SENDGRID_PASSWORD'],
+  domain: ENV['APP_DOMAIN'],
+  address: 'smtp.sendgrid.net',
+  port: 587,
+  authentication: :plain,
+  enable_starttls_auto: true
+}
+```
+
+- Add the ENV variable `APP_DOMAIN` to the testing, develop and master project on heroku.
+
+- Add a Domain Whitelabel at <https://app.sendgrid.com/settings/whitelabel/domains>
+
```

Comment:
The part regarding the creation of an account on sendgrid is missing. which rules do we have to follow?